array-manipulation-3/src/zip.js

```diff
@@ -11,21 +11,19 @@
 function zip(first, second) {
   var output = [];
   var array = [];
+  var length = Infinity;
 
-  for (var i = 0; i < first.length; i++) {
-    if (first.length === second.length) {
-      array.push(first[i]);
-      array.push(second[i]);
-      output.push(array);
-      array = [];
-    } else {
-      if (second[i] !== undefined) {
-        array.push(first[i]);
-        array.push(second[i]);
-        output.push(array);
-        array = [];
-      }
-    }
+  if (first.length < length) {
+    length = first.length;
+  }
+  if (second.length < length) {
+    length = second.length;
+  }
```

Comment:
If their lengths are equal, you are going to loop until infinity 🤢

Reply:
Hi Cody, not sure what you mean. Some of the test arrays on index.html had equal lengths and it passed. Also, if length = Infinity, wouldn't first.length always be less?